include/seqan3/alphabet/composite/alphabet_variant.hpp

```diff
@@ -403,13 +421,15 @@ class alphabet_variant :
      * \stableapi{Since version 3.1.}
      */
     template <typename alphabet_variant_t, typename indirect_alternative_type>
-    friend constexpr auto
-    operator==(alphabet_variant_t const lhs, indirect_alternative_type const rhs) noexcept -> std::enable_if_t<
-        detail::variant_comparison_guard<alphabet_variant_t, indirect_alternative_type, false, alternative_types...>,
-        bool>
+        requires (!detail::is_span<alphabet_variant_t>::value)
+              && (detail::variant_comparison_guard<alphabet_variant_t,
+                                                   indirect_alternative_type,
+                                                   false,
+                                                   alternative_types...>)
+    friend constexpr bool operator==(alphabet_variant_t const lhs, indirect_alternative_type const rhs) noexcept
     {
         using alternative_predicate = detail::weakly_equality_comparable_with_<indirect_alternative_type>;
-        constexpr auto alternative_position =
+        constexpr bool alternative_position =
```

Comment:
This looks suspicious. isn't `alternative_position` holding the position?

Reply:
Yes, good catch. That was a search/replace error when changing the return type of two operators from `auto .... -> bool` to `bool` :)